auth/CognitoSecureStorage.ts

```diff
@@ -36,109 +37,121 @@ const STORE_CHUNK_SIZE = 2048
+  async removeItem (key: string) {
+    const keyChunkMappings = await this.loadKeyChunkMappings()
+    if (keyChunkMappings) {
+      const amountOfChunks = keyChunkMappings.get(key) ?? 0
+      await Promise.allSettled(
+        ArrayUtils.repeatElements(amountOfChunks, async (chunkIndex) => {
+          await this.store.deleteItemAsync(this.secureStoreKey(key, chunkIndex))
+        })
+      )
+      keyChunkMappings.delete(key)
     }
+    this.keyChunkMappings = keyChunkMappings
+    await this.saveKeyChunkMappings()
   }
 
-  getItem (key: string) {
-    return this.cache.get(key)
+  async setItem (key: string, value: string) {
+    const keyChunkMappings = await this.loadKeyChunkMappings()
+    if (keyChunkMappings) {
+      const chunks = this.stringToChunks(value)
+      for (let i = 0; i < chunks.length; i++) {
+        const chunkKey = this.secureStoreKey(key, i)
+        await this.store.setItemAsync(chunkKey, chunks[i])
+      }
+      keyChunkMappings.set(key, chunks.length)
+      this.keyChunkMappings = keyChunkMappings
+      await this.saveKeyChunkMappings()
+    }
   }
 
-  removeItem (key: string) {
-    this.cache.delete(key)
-
-    const amountOfChunks = this.keyChunkMappings.get(key) ?? 0
-    for (let i = 0; i < amountOfChunks; i++) {
-      this.store.deleteItemAsync(secureStoreKey(key, i))
+  async getItem (key: string) {
+    const keyChunkMappings = await this.loadKeyChunkMappings()
+    if (keyChunkMappings.size > 0) {
+      const amountOfChunks = keyChunkMappings.get(key) ?? 0
+      const chunks = [] as string[]
+      for (let i = 0; i < amountOfChunks; i++) {
+        const chunk = await this.store.getItemAsync(this.secureStoreKey(key, i))
+        if (chunk) {
+          chunks.push(chunk)
+        }
+      }
+      if (chunks.length === 0) {
+        return null
+      }
+      return chunks.join("")
+    } else {
+      return null
     }
-
-    this.keyChunkMappings.delete(key)
-    this.saveKeyChunkMappings()
   }
 
   async clear () {
+    const keyChunkMappings = await this.loadKeyChunkMappings()
     await Promise.allSettled(
-      Array.from(this.keyChunkMappings).map(async ([key, amountOfChunks]) => {
+      Array.from(keyChunkMappings).map(async ([key, amountOfChunks]) => {
         await Promise.allSettled(
           ArrayUtils.repeatElements(amountOfChunks, async (chunkIndex) => {
-            await this.store.deleteItemAsync(secureStoreKey(key, chunkIndex))
+            await this.store.deleteItemAsync(
+              this.secureStoreKey(key, chunkIndex)
+            )
           })
         )
       })
     )
-    this.keyChunkMappings = new Map()
+    this.keyChunkMappings.clear()
     await this.saveKeyChunkMappings()
-    this.cache = new Map()
   }
 
-  async sync () {
-    if (this.syncPromise) {
-      return await this.syncPromise
+  private async loadKeyChunkMappings () {
+    if (this.keyChunkMappingsPromise) {
+      return await this.keyChunkMappingsPromise
```

Comment:
I'm pretty sure that this would load outdated chunks and thus cause the get/set/remove methods to have incorrect behavior. There should be another check to see if we have `this.keyChunkMappings` as not undefined and just return that in that case. The reason I don't think you've done this however is because there's no test cases that expose this flaw, which would be doable by first doing some operations in `testStorage1` and then doing some operations (and not just calling `getItem`) in `testStorage2`.